### PR TITLE
tools/esp32: Add espressif esptool as a submodule

### PR DESCRIPTION
**Pull request by zhouxinhe**
esptool is under GPL license.
we add it as a submodule.
@xiarihf @xixidodo

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [99373da6b208eab46d6f8c07b58eb153d48bf34c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [99373da6b208eab46d6f8c07b58eb153d48bf34c] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@zhouxinhe Does it, submodule resolve a GPL issue? Even if it is a part of submodule, if it makes some header of binary or leaves some evidence, it will be a problem. Could you confirm it?

**Comment by btheosam** [comment]
@zhouxinhe,
Are supporting secure boot now? Is espsecure.py a immediate need? 
Is is possible to get the download tools (under another license) from expressif?

File: build/tools/esp32/esptool_py/Makefile [hunk 1]
```diff
@@ -46,13 +46,13 @@ APP_BIN:=$(APP_ELF:.elf=.bin)
...
 #
-ESPTOOLPY_SRC := $(CURRENTDIR)/esptool_py/esptool.py
+ESPTOOLPY_SRC := $(CURRENTDIR)/esptool_py/esptool/esptool.py
 ESPTOOLPY := $(PYTHON) $(ESPTOOLPY_SRC) --chip esp32
 ESPTOOLPY_SERIAL := $(ESPTOOLPY) --port $(ESPPORT) --baud $(ESPBAUD) --before $(CONFIG_ESPTOOLPY_BEFORE) --after $(CONFIG_ESPTOOLPY_AFTER)
 
 # Supporting esptool command line tools
-ESPEFUSEPY := $(PYTHON) $(CURRENTDIR)/esptool/espefuse.py
-ESPSECUREPY := $(PYTHON) $(CURRENTDIR)/esptool/espsecure.py
+ESPEFUSEPY := $(PYTHON) $(CURRENTDIR)/esptool_py/esptool/espefuse.py
```
Comment (xiarihf) [inline build/tools/esp32/esptool_py/Makefile]: Useless definition in Makefiles.

**Comment by seinfra** [comment]
Target : [63134ef08b28d2e3cf8af11fa7353bf61d037643] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [63134ef08b28d2e3cf8af11fa7353bf61d037643] - Code Rule Check OK.

**Comment by zhouxinhe** [comment]
> @zhouxinhe Does it, submodule resolve a GPL issue? Even if it is a part of submodule, if it makes some header of binary or leaves some evidence, it will be a problem. Could you confirm it?

- this esptool.py is only used to download binary to board, will not make any header in binary.
- [esp-idf](https://github.com/espressif/esp-idf/tree/master/components/esptool_py) (Apache License 2.0) also has a reference of esptool.

File: build/tools/esp32/esptool_py/Makefile (hunk 1 above)
Comment (zhouxinhe) [reply build/tools/esp32/esptool_py/Makefile]: > @zhouxinhe,
> Are supporting secure boot now? Is espsecure.py a immediate need?
> Is is possible to get the download tools (under another license) from expressif?

- secure boot is not supported now, I deleted definitions related to espsecure.py&espefuse.py.
- there's download tools espressif provided, @xiarihf is verifying it and will share with you later.

**Comment by zhouxinhe** [comment]
> @zhouxinhe,
> Are supporting secure boot now? Is espsecure.py a immediate need?
> Is is possible to get the download tools (under another license) from expressif?

- secure boot is not supported now, I deleted definitions related to espsecure.py&espefuse.py.
- there's download tools espressif provided, @xiarihf is verifying it and will share with you later.

File: build/tools/esp32/esptool_py/Makefile (hunk 1 above)
Comment (zhouxinhe) [reply build/tools/esp32/esptool_py/Makefile]: Fixed

**Pull request merged**